File: C/Project Euler #037.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int isPrime(int n);
int left(int n);
int right(int n);

/* Solución para el problema #37 de Project Euler
 * "Encuentro la suma de los únicos once números primos que son a la vez truncables de izquierda a derecha y de derecha a izquierda."
 * https://projecteuler.net/problem=37 */
 
int main()
{
    int i, j = 10, suma;
    while (i < 11){
        if ((left(j) == 1) && (right(j) == 1)){
            printf("%d\n", j);
            suma += j;
            i++;
        }
        j++;
    }
    printf("La respuesta es %d\n", suma);
    printf("Fin del programa\n");
    return 0;
}

int left(int n){ //Desde la izquierda
    int i = 1, j;
    while (i < n) i *= 10;
    while (i >= 1){
        j = n % i;
        if (isPrime(j) == 0) return 0;
        i /= 10;
    }
    if (isPrime(j) == 1) return 1;
    return 0;
}

int right(int n){ //Desde la derecha
    int i = 1, j;
    while (i < n){
        j = n / i;
        if (isPrime(j) == 0) return 0;
        i *= 10;
    }
    if (isPrime(j) == 1) return 1;
    return 0;
}

int isPrime(int n){//Función para determinar si un numero es primo, basada en la Criba de Eratóstenes
    if (n == 1) return 0;
    else if (n < 4) return 1; //Si son menores de 4, son primos
    else if (n % 2 == 0) return 0; //Si son pares, no son primos
    else if (n < 9) return 1; //Si son menores de 9 e impares, son primos
    else if (n % 3 == 0) return 0; //Si son múltiplos de 3, no son primos
    else{ 
        int r = floor(sqrt(n));
        int f = 5;
        while (f <= r){
            if (n % f == 0) return 0;
            else if (n % (f + 2) == 0) return 0;
            else f += 6;
        }
        return 1;
    }
}

File: Visual Basic .NET/Project Euler #037.vb
Module Module1
    
    ' Solución para el problema #37 de Project Euler
    ' "Encuentra la suma de los once únicos números primos que son truncables de izquierda a derecha y de derecha a izquierda a la vez"
    ' https://projecteuler.net/problem=37
    ' Código espagueti de 2016; optimizar
    
    Function isPrime(ByVal n As Integer) As Boolean 'Función para determinar si un numero es primo.
        If n = 1 Then
            Return False
        ElseIf n < 4 Then
            Return True
        ElseIf n Mod 2 = 0 Then
            Return False
        ElseIf n < 9 Then
            Return True
        ElseIf n Mod 3 = 0 Then
            Return False
        Else
            Dim r As Integer = Math.Floor(Math.Sqrt(n))
            Dim f As Integer = 5
            Do While f <= r
                If n Mod f = 0 Then
                    Return False
                ElseIf n Mod (f + 2) = 0 Then
                    Return False
                Else
                    f += 6
                End If
            Loop
            Return True
        End If
    End Function
    Sub Main()

        Dim n As Integer = Nothing
        Dim n2 As Integer = Nothing
        Dim digits() As Char = {"0", "0", "0", "0", "0", "0", "0", "0", "0"}
        Do
            n += 1
            If isPrime(n) And n > 9 Then
                Dim trucatable As Integer = n
                Dim prime As String = n.ToString
                Dim isTrucatable As Boolean = True
                digits = prime.ToCharArray()
                n2 = 0
                prime = Nothing
                Do Until trucatable < 9 Or isTrucatable = False
                    digits(n2) = Nothing
                    Do Until digits(n2) = Nothing
                        prime += digits(n2)
                        n2 += 1
                    Loop
                    Do Until digits(n2) = Nothing
                        n -= 1
                    Loop
                    n2 -= 1
                    trucatable = Integer.Parse(prime)
                    If isPrime(trucatable) = False Then
                        isTrucatable = False
                    End If
                    Console.Write(n & "   " & trucatable)
                    Console.ReadLine()
                    n2 += 1
                Loop
                digits = prime.ToCharArray()
                prime = Nothing
                n2 = 1
                Do Until trucatable < 9 Or isTrucatable = False
                    digits(digits.Length - n2) = Nothing
                    'trucatable = Integer.Parse(digits(0) & digits(1) & digits(2) & digits(3) & digits(4) & digits(5) & digits(6) & digits(7) & digits(8))
                    If isPrime(trucatable) = False Then
                        isTrucatable = False
                    End If
                    n2 += 1
                Loop
                Dim respuesta = Nothing
                If isPrime(trucatable) And trucatable < 9 Then
                    respuesta += n
                    Console.Write(n & "   " & trucatable & "   " & respuesta)
                    Console.ReadLine()
                End If
            End If
        Loop

    End Sub
End Module
